lib/base-compiler.js

```diff
@@ -1733,7 +1733,7 @@ Please select another pass or change filters.`;
     // to simple attempts with a clear diagnostic; the service still needs to
     // assume that malicious actors can make the compiler open arbitrary files.
     checkSource(source) {
-        const re = /^\s*#\s*i(nclude|mport)(_next)?\s+["<](\/|.*\.\.)[">]/;
+        const re = /^\s*#\s*i(nclude|mport)(_next)?\s+["<]((\.{1,2}|\/)[^">]*)[">]/;
```

Comment:
`./blah`
`../bleh`
`/blep`

should all be covered here. The `[^">]` part comes from https://github.com/compiler-explorer/compiler-explorer/blob/main/static/compiler-service.js#L270